### PR TITLE
Fix various errors in deadzone calculation/filtering

### PR DESCRIPTION
**Pull request by ndarilek**
I noticed that a number of deadzones weren't correctly being calculated. In particular, inputs with values like (1, 0.05) and a deadzone of 0.1 would return (1, 0.05) for the axis values when I believe they should have returned (1, 0). I added initially failing tests that demonstrated this behavior.

Additionally, calculating whether a dual-axis input was pressed or released simply looked at its value;. It didn't check whether each member axis was above the threshold.

Finally, there's some confusion in my mind about `positive_low` and `negative_low`. In particular I think `negative_low` should always be negative, tested by assertion. `SingleAxis::negative_only` was just passing the same deadzone in for both values and failing the additional calculations I'd added. I struggled a bit with this until I discovered I was passing a positive value in for the threshold. Fixing that got my controls working as I'd expect, but ideally the API should prevent this.

I think this PR is an improvement, but it may not correctly represent the intent behind `negative_low`. If it should always be positive and negated for testing, I'd like to see assertions somewhere so the intent is clear. Happy to add those here or elsewhere, but since structs can be created from anywhere, I'm not immediately sure what the best place for that check would be.

Thanks.

### CODE REVIEW

**Comment by ndarilek** [comment]
Hey, just wanted to check on this one.

Also, contributing.md mentions development happening on the dev branch, but it looks like that hasn't been updated since last year? Just want to make sure I'm not missing something.

I've been testing this branch and it seems fine, but it's possible I've missed something or made a mistake. Hoping to cut a release of my game with it on Monday or Tuesday.

Thanks.

**Comment by alice-i-cecile** [comment]
That's correct: that practice has been abandoned. Let me review this PR now; I've been a little burnt out <3

**Review by alice-i-cecile — APPROVED** [review]
This LGTM: those are both bugs. Can you update CHANGELOG.md briefly to reflect your changes?

**Comment by ndarilek** [comment]
Sure thing. Sorry, been there re: burnout and didn't mean to rush you.

**Comment by ndarilek** [comment]
Done. There wasn't a top-level section for the new release so I just added the update (hopefully) correctly-structured and near where it should go. Hope that's sufficient.

Thank you!

File: RELEASES.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Release Notes
 
+### Bugs
```
Comment (alice-i-cecile) [inline RELEASES.md:3]:
```suggestion

## Unreleased

### Bugs

```

**Comment by alice-i-cecile** [comment]
Just a minor formatting nit, but I've fixed that up for you. Merging!

**Pull request merged**